reference/reference.go

```diff
@@ -178,10 +184,26 @@ func Parse(s string) (Reference, error) {
 	return r, nil
 }
 
-// ParseNamed parses the input string and returns a named
-// object representing the given string. If the input is
-// invalid ErrReferenceInvalidFormat will be returned.
-func ParseNamed(name string) (Named, error) {
+// ParseNamed parses s and returns a syntactically valid reference implementing
+// the Named interface. The reference must have a name, otherwise an error is
+// returned.
+// If an error was encountered it is returned, along with a nil Reference.
+// NOTE: ParseNamed will not handle short digests.
+func ParseNamed(s string) (Named, error) {
+	ref, err := Parse(s)
+	if err != nil {
+		return nil, err
+	}
+	named, isNamed := ref.(Named)
+	if !isNamed {
+		return nil, fmt.Errorf("reference %s has no name", ref.String())
+	}
+	return named, nil
+}
+
+// WithNamed returns a named object representing the given string. If the input
+// is invalid ErrReferenceInvalidFormat will be returned.
+func WithNamed(name string) (Named, error) {
```

Comment:
> Would expect WithName instead of WithNamed. WithNamed would imply it was attaching a Named to a Reference instead of just getting a reference with a name (string).

Oops, fixed.